src/IDisposableGenerator/WorkItemCollection.cs

```diff
@@ -49,16 +49,30 @@ public List<WorkItem> GetWorkItems()
     public int IndexOf(WorkItem item)
         => this.WorkItems.IndexOf(item);
 
-    private static ClassItems GetClassItem(AttributeData attr, INamedTypeSymbol testClass)
+    private static ClassItems? GetClassItem(INamedTypeSymbol testClass)
     {
-        var result = new ClassItems
+        var result = new ClassItems();
+        var hasDisposalGeneration = false;
+
+        foreach (var attr in testClass.GetAttributes())
         {
-            Name = testClass.Name,
-            Accessibility = testClass.DeclaredAccessibility,
-            Stream = (bool)attr.ConstructorArguments[0].Value!,
-        };
+#pragma warning disable IDE0010 // Add missing cases
+            switch (attr.AttributeClass!.Name)
+            {
+                case "GenerateDisposeAttribute":
+                    hasDisposalGeneration = true;
+                    result.Name = testClass.Name;
+                    result.Accessibility = testClass.DeclaredAccessibility;
+                    result.Stream = (bool)attr.ConstructorArguments[0].Value!;
+                    break;
+                case "GenerateThrowIfDisposedAttribute":
```

Comment:
It sounds like opting out of the feature should be explicit instead of opting in. I agree that promoting the usage of the throw helper is better, and I'll invert the logic of the attribute.